latest version (#4887)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/deps.md

<VersionBlock firstVersion="1.7">

> ## Predictable package installs

Author: sugg: shorter title so people can scan faster

Reviewer: good call!

> Starting in dbt Core v1.7, dbt generates an additional `package-lock.yml` file in the root of your project. This contains the full set of resolved packages, based on the `packages` configuration (in `dependencies.yml` or `packages.yml`). Each subsequent invocation of `dbt deps` will install from the _locked_ set of packages specified in this file. By storing the full set of required packages (with pinned versions) in version-controlled code, it ensures predictable installs in production and consistency across all developers and environments.

Reviewer 2: Suggest removing "additional" so it doesn't read like there will be 2 package-lock.yml files in the project
```suggestion
Starting in dbt Core v1.7, dbt generates a `package-lock.yml` file in the root of your project. This contains the complete set of resolved packages based on the `packages` configuration in `dependencies.yml` or `packages.yml`. Each subsequent invocation of `dbt deps` will install from the _locked_ set of packages specified in this file. Storing the complete set of required packages (with pinned versions) in version-controlled code ensures predictable installs in production and consistency across all developers and environments.
```

> The `package-lock.yml` file includes a `sha1_hash` of the `packages` config. This enables dbt to detect if the `packages` config has updated, and to rerun dependency resolution. If you want to only check for changes to the `packages` config and update the lock file accordingly (without actually installing those packages), you can provide the `--lock` option (that is, `dbt deps --lock`).

Reviewer 2: ```suggestion
The `package-lock.yml` file includes a `sha1_hash` of the `packages` config. This enables dbt to detect if the `packages` config has been updated, and to rerun dependency resolution. To only check for changes to the `packages` config and update the lock file accordingly without installing those packages, provide the `--lock` flag (that is, `dbt deps --lock`).
```

> It's also possible to force package resolution to rerun, even if the `packages` config hasn't changed, by running `dbt deps --upgrade`. For example, this enables you to always get the latest commits from the `main` branch of an internally maintained `git` package while accepting the risk of unpredictable builds. An alternative to always running `dbt deps --upgrade` in production is to "ignore" the lock file by adding `package-lock.yml` to your project's `.gitignore` file. If you pursue either approach, we strongly recommend adding version pins for third-party packages within your `packages` config.

Reviewer 2: ```suggestion
It's possible to force package resolution to rerun, even if the `packages` config hasn't changed, by running `dbt deps --upgrade`. This enables you to get the latest commits from the `main` branch of an internally maintained `git` package while accepting the risk of unpredictable builds. An alternative to running `dbt deps --upgrade` in production is to "ignore" the lock file by adding `package-lock.yml` to your project's `.gitignore` file. If you pursue either approach, dbt Labs strongly recommends adding version pins for third-party packages within your `packages` config.
```

## Add specific packages

> You can use the `dbt deps` command to add or update an existing package configuration &mdash; no need to remember the exact syntax for package configurations. 

Reviewer 2: ```suggestion
The `dbt deps` command can add or update an existing package configuration &mdash; no need to remember the exact syntax for package configurations. 
```

For Hub packages (default), which are the easiest to install:

```shell
dbt deps --add-package dbt-labs/dbt_utils@1.0.0

# with semantic version range
dbt deps --add-package dbt-labs/snowplow@">=0.7.0,<0.8.0"
```

> For other package types, use the `--source` option:

Reviewer 2: ```suggestion
For other package types, use the `--source` flag:
```

> For Git or local packages, use the `--source` option:

Author: ```suggestion
For other package types, use the `--source` option:
```

> ```shell

Author: improved example chunking for faster scanning

Reviewer: I like it. For this line in particular, how about just:
```
For other package types, use the `--source` option:
```
